src/Microsoft.Security.Utilities.Core/PreciselyClassifiedSecurityKeys/SEC101_565_SecretScanningSampleToken.cs

```diff
@@ -23,6 +18,15 @@ public SecretScanningSampleToken()
             Signatures = new HashSet<string>(["ab85"]);
         }
 
+#if HIGH_PERFORMANCE_CODEGEN
+        IEnumerable<HighPerformancePattern> IHighPerformanceScannableKey.HighPerformancePatterns => [
+            new(signature: "ab85",
+                scopedRegex: "^secret_scanning_.{4}fc6f8d7638cf1c11da812da308d43_[0-9A-Za-z]{5}",
```

Comment:
>.{4} [](http://example.com/codeflow?start=47&length=4)

why are you permitting any characters here?

Reply:
Because the scoped regex only runs if the signature has matched already. It's a bit of a micro-optimization here, but in other cases this lets us share regexes that would differ only by signature.

Reply:
OK, so in general, you are saying this is a good practice that futures proofs allowing us to share regexes.

A different lens would say do *not* add this loosening unless/until there is regex sharing. certainly there shouldn't be for this rule (and several others with the pattern). 

but the other (small) part of your case is that generalizing this piece of the regex allows a microoptimization in the regex itself? i.e., simplifying it/making it faster (in some non-zero but potentially small way)?

Reply:
Yes, to the last (small) part, you can actually see the code getting improved if you go-to-definition on the methods with GeneratedRegexAttribute.

My inclination here was to keep all of the high-performance patterns consistent in skipping the signature rather than make individual judgment calls. I don't feel super strongly about it, but that still feels better to me.

Reply:
no strong feeling here, defer to code author.